skywalking/plugins/sw_sanic.py

```diff
@@ -0,0 +1,89 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+
+from skywalking import Layer, Component, config
+from skywalking.trace import tags
+from skywalking.trace.carrier import Carrier
+from skywalking.trace.context import get_context
+from skywalking.trace.span import NoopSpan
+from skywalking.trace.tags import Tag
+
+logger = logging.getLogger(__name__)
+
+version_rule = {
+    "name": "sanic",
+    "rules": [">=20.3.0"]
+}
+
+_status_code = 0
+
+
+def install():
+    from inspect import isawaitable
+    from sanic import Sanic, handlers, response
+
+    _format_http1_response = response.format_http1_response
+    _handle_request = Sanic.handle_request
+    _handlers_ErrorHandler_reponse = handlers.ErrorHandler.response
+
+    def _sw_format_http1_reponse(status: int, headers, body=b""):
+        global _status_code
+        _status_code = status
+        return _format_http1_response(status, headers, body)
```

Comment:
Please don't use a global variable to store the status code, in concurrent scenario, the status code may mess up in different requests.

You can use the same way as line 81 to get the current active span (nullablity check) and set the status code here

```python
entry_span = get_context().active_span()
```

Reply:
ok,thanks.i'll try to fix it later